sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -142,24 +77,34 @@ class usm_allocator {
   ///
   /// \param Ptr is a pointer to memory being deallocated.
   /// \param Size is a number of elements previously passed to allocate.
-  void deallocate(pointer Ptr, size_t) {
+  void deallocate(T *Ptr, size_t) {
     if (Ptr) {
       free(Ptr, MContext);
     }
   }
 
+  template <class U, usm::alloc AllocKindU, size_t AlignmentU>
+  friend bool operator==(const usm_allocator<T, AllocKind, Alignment> &One,
+                         const usm_allocator<U, AllocKindU, AlignmentU> &Two) {
+    return ((AllocKind == AllocKindU) && (One.MContext == Two.MContext) &&
```

Comment:
There is no need to have this templated. Comparison between different allocator types aren't needed.
```suggestion
  friend bool operator==(const usm_allocator &One,
                         const usm_allocator &Two) {
    return ((One.MContext == Two.MContext) &&
```

If you do want to keep the templated version would also need to check that `is_same_v<T, U> && AlignmentU==Alignment`. Same for next one.

Reply:
This was the only way I could get allocate_shared to work.

Reply:
What's the error you get without it? It makes no sense this should be needed because it isn't a requirement.